addon/ajax-request.js

```diff
@@ -47,6 +48,14 @@ export default class AjaxRequest {
       type: hash.type,
       url: hash.url
     };
+
+    const allHeaders = this._getFullHeadersHash(hash.headers);
+    if (allHeaders['Content-Type'] === JSONAPIContentType) {
```

Comment:
This doesn't allow for JSONAPI requests with an extension. When an extension is used, the extension name is added to the end of the content type. For example: `Content-Type: application/vnd.api+json; ext="ext1,ext2";`

We could do something like
```
function isJSONAPIContentType(header) {
  return header.startsWith('application/vnd.api+json');
}
```